dotnet/src/Connectors/Connectors.AI.OpenAI/OpenAIKernelBuilderExtensions.cs

```diff
@@ -49,9 +50,9 @@ public static class OpenAIKernelBuilderExtensions
         bool setAsDefault = false,
         HttpClient? httpClient = null)
     {
-        builder.WithAIService<ITextCompletion>(serviceId, (loggerFactory, config) =>
+        builder.WithAIService<ITextCompletion>(serviceId, (loggerFactory) =>
         {
-            var client = CreateAzureOpenAIClient(loggerFactory, config, deploymentName, endpoint, new AzureKeyCredential(apiKey), httpClient);
+            var client = CreateAzureOpenAIClient(loggerFactory, builder.HttpHandlerFactory, deploymentName, endpoint, new AzureKeyCredential(apiKey), httpClient);
             return new AzureTextCompletion(deploymentName, client, loggerFactory);
         }, setAsDefault);
```

Comment:
I'm not sure about this pattern: `builder.WithAIService(() => ...builder.HttpHandlerFactory...)`. 
Shouldn't it be like this?
```csharp
builder.WithAIService<ITextCompletion>(serviceId, (loggerFactory, httpHandlerFactory) =>
{
    var client = CreateAzureOpenAIClient(loggerFactory, httpHandlerFactory, deploymentName, endpoint, new AzureKeyCredential(apiKey), httpClient);
    return new AzureTextCompletion(deploymentName, client, loggerFactory);
}, setAsDefault);
```

Same applies to all other extension methods. In this case you don't need to expose `HttpHandlerFactory` from `KernelBuilder`.

Reply:
Completely agree, missed this totally. Thanks for the review!